infrastructure/quick-deploy/aws/ecr.tf

```
@@ -1,12 +1,12 @@
 locals {
-  mongodb_image_name = can(coalesce(var.mongodb_sharding)) ? coalesce(var.mongodb.image_name, "bitnami/mongodb-sharded") : coalesce(var.mongodb.image_name, "bitnami/mongodb")
+  mongodb_image_name = var.mongodb_sharding != null ? (var.mongodb != null ? coalesce(var.mongodb.image_name, "bitnami/mongodb-sharded") : "bitnami/mongodb-sharded") : (var.mongodb != null ? coalesce(var.mongodb.image_name, "bitnami/mongodb") : "bitnami/mongodb")
   ecr_input_images = concat([
     [var.eks.docker_images.cluster_autoscaler.image, var.eks.docker_images.cluster_autoscaler.tag],
     [var.eks.docker_images.instance_refresh.image, var.eks.docker_images.instance_refresh.tag],
     [var.metrics_server.image_name, var.metrics_server.image_tag],
     [var.keda.keda_image_name, var.keda.keda_image_tag],
     [var.keda.apiserver_image_name, var.keda.apiserver_image_tag],
-    [local.mongodb_image_name, var.mongodb.image_tag],
+    [local.mongodb_image_name, var.mongodb != null ? var.mongodb.image_tag : "latest"],
```

This supports `null` elements, so you can just write the following:

```suggestion
    var.mongodb != null ? [local.mongodb_image_name,  var.mongodb.image_tag] : null,
```

This has the added benefit to avoid upload the mongodb image if it is not deployed.